### DISCUSSION TITLE
2020 ends: happy new year AzerothCore!

### DISCUSSION THREAD

**Discussion by FrancescoBorzi** [azerothcore/azerothcore-wotlk#4123] [News & Blog]
Dear AzerothCore community,

2020 has not been a good year for the world out there, but it definitely has been a great year for our project.

We are the biggest open-source WoW emulation community of 2020. During the year, we pushed improvements to the master branch every day.
New contributors have joined the project, and several old ones came back. We fixed plenty of game contents and added new features.
We introduced unit-testing. We made our application more stable, fixing some critical issues. We kept modernizing our codebase by replacing legacy code with modern, clean C++.
We completed Keira3 (started in mid-2019), providing the best WoW world-Database editor available in the market, and tailored it specifically for AzerothCore.

We improved our CI by adding many checks which will make impossible for new PRs to break the build or add new code warnings on all major platforms and compilers
(I firmly believe that this, together with our dry-run and DB errors check, makes our CI the best you can find in WoW emulation).
We improved the pathfinding mechanism. We added tons of translation strings to improve the gaming experience on non-English clients.
We added plenty of modules and custom tools too, because the world is beautiful because it is varied!
We did many other improvements that are just too many to be mentioned here.
Finally, we launched the child-project ChromieCraft aimed to perfection all game contents from A to Z and it's going very well.
When I say "we", of course I mean the AzerothCore community. Because none of the above would have been possible without such a strong, friendly and collaborative community.

So I wish to thank you all: whoever you are, wherever you come from, even if you just contributed to AzerothCore by testing one PR, reporting or confirming a small bug, changing one line of code, improving a small text in our wiki, creating some nice tool for AzerothCore, giving help to another user, etc... :  **THANK YOU** .
With your efforts, made this place better than it was.

I'm sure 2021 will be another great year of our community and I'm looking forward to seeing it grow even more!
Stay safe and have a great new year :partying_face: :partying_face: :partying_face: .

Yours sincerely,
 -- Shin aka Francesco

**Comment by yxylwt** [comment]
happy new year
